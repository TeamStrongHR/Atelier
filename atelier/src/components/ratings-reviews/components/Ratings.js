import {useState, useEffect} from 'react';
import StarRating from '../../shared/StarRating.js';
import axios from 'axios';
export default function Ratings ({breakdown}) {

  //math to determine average rating...
  var totalNumOfRatings = 0;
  var total = 0;
  for(const num in breakdown.ratings) {
    var val = parseInt(num);
    var count = parseInt(breakdown.ratings[num]);
    totalNumOfRatings+=count;
    total+=(val*count)
  }

  var averageRating = (((total/parseFloat(totalNumOfRatings)) * 100)/100).toFixed(1)


  //for use in printing out the progress bars
  //reverse to start array from 5 stars...
  var ratingsKeys = Object.keys(breakdown.ratings).reverse();
  var ratingsValues = Object.values(breakdown.ratings).reverse();
  var recommended = Math.round(parseInt(breakdown.recommended.true)/parseFloat((parseInt(breakdown.recommended.true)+ parseInt(breakdown.recommended.false)))*100);


  return (
    <div className="ratings">
      <h5>RATINGS & REVIEWS</h5>

      {/*RATING STARS*/}
      <div id="num-stars">
        <span id="num"><strong>{averageRating}</strong></span>
        <StarRating rating={averageRating}/>
      </div>
      <p>{recommended}% of reviews recommend this product</p>

      {/*RATING BARS*/}
      {ratingsKeys.map((key, i) => {
        return (
        <div className="progress-bar">
        <div><u>{key} stars</u></div>
        <div className="whole-bar">
          <div className="partial-bar" style={{width:`${Math.round((parseInt(ratingsValues[i])/parseFloat(totalNumOfRatings))*100)}%`}}></div>
        </div>
      </div>)
      })}

    {/*RANGES*/}
    <div className="range">
      {Object.keys(breakdown.characteristics).map(key => {
        console.log(breakdown.characteristics, key, ((breakdown.characteristics[key].value)/parseFloat(5))*100);
        return (<div className="characteristics">
          <h6>{key}</h6>
          <input type="range" min="0" max="100" value={((breakdown.characteristics[key].value)/parseFloat(5))*100} list="ticks-size"/>
          <datalist id="ticks-size">
            <option value="33"></option>
            <option value="66"></option>
          </datalist>
        </div>)
      })}
    </div>
    </div>
  )
}

