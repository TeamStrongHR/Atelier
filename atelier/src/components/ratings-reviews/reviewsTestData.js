export var reviews = {
  "product": "37313",
  "page": 0,
  "count": 5,
  "results": [
      {
          "review_id": 1277202,
          "rating": 3,
          "summary": "Test Submit",
          "recommend": false,
          "response": null,
          "body": "Apsafdjkabflkjhbsdkjafhbdkjlhbflkdsabkvldbajkvavdfsdvsdvds",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "Rendy",
          "helpfulness": 3,
          "photos": [
              {
                  "id": 2456480,
                  "url": "https://res.cloudinary.com/dvijvlkad/image/upload/ja6cv28co6xuyb1nvk9k"
              }
          ]
      },
      {
          "review_id": 1277203,
          "rating": 3,
          "summary": "Test Submit",
          "recommend": false,
          "response": null,
          "body": "Apsafdjkabflkjhbsdkjafhbdkjlhbflkdsabkvldbajkvavdfsdvsdvds",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "Rendy",
          "helpfulness": 2,
          "photos": [
              {
                  "id": 2456481,
                  "url": "https://res.cloudinary.com/dvijvlkad/image/upload/ja6cv28co6xuyb1nvk9k"
              }
          ]
      },
      {
          "review_id": 1277205,
          "rating": 3,
          "summary": "HAPPY HAPPY HAPPY",
          "recommend": false,
          "response": null,
          "body": "THIS PRODUCT IS AWESOME WOOHOOO THIS IS AWESOME",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "Ray",
          "helpfulness": 2,
          "photos": []
      },
      {
          "review_id": 1277207,
          "rating": 3,
          "summary": "gfdsfdsgasdgdsagsdgsa",
          "recommend": true,
          "response": null,
          "body": "fdasasfasgadgdsagdgdfsaaaaaaaaaaaaaaaaaaaaa",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "gsagdsag",
          "helpfulness": 1,
          "photos": []
      },
      {
          "review_id": 1277206,
          "rating": 4,
          "summary": "Whats up hahahaha",
          "recommend": false,
          "response": null,
          "body": "This product is super duper cool and i lkove it ",
          "date": "2022-10-25T00:00:00.000Z",
          "reviewer_name": "Rendy",
          "helpfulness": 1,
          "photos": []
      }
  ]
}

export var breakdown = {
  "product_id": "37313",
  "ratings": {
      "1": "5",
      "2": "3",
      "3": "18",
      "4": "24",
      "5": "25"
  },
  "recommended": {
      "false": "16",
      "true": "59"
  },
  "characteristics": {
      "Fit": {
          "id": 125036,
          "value": "2.6666666666666667"
      },
      "Length": {
          "id": 125037,
          "value": "2.7903225806451613"
      },
      "Comfort": {
          "id": 125038,
          "value": "2.8730158730158730"
      },
      "Quality": {
          "id": 125039,
          "value": "3.1269841269841270"
      }
  }
}
