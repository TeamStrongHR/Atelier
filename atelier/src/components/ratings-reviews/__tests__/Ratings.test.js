import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Ratings from '../components/Ratings.js';
import {reviews, breakdown} from '../reviewsTestData.js';


afterEach(() => {
  cleanup()
});



describe("Ratings Component", ()=> {
  render(<Ratings breakdown={breakdown} />);
    const container = screen.getByTestId('ratings-comp')

  test("Should render Ratings component", ()=> {
    expect(container).toBeInTheDocument();
  })

  test("Should have '% of reviews recommend this product'", () => {
    expect(container).toHaveTextContent('% of reviews recommend this product')
  })

  test("It should render StarRating component", ()=> {
    expect(container.getElementsByClassName("star-rating")).toBeDefined();
  })

  test('matches snapshot', ()=> {
    const tree = renderer.create(<Ratings breakdown={breakdown}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
