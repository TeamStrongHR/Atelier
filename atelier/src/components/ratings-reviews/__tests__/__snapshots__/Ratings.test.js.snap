// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings Component matches snapshot 1`] = `
<div
  className="ratings"
  data-testid="ratings-comp"
>
  <h5>
    RATINGS & REVIEWS
  </h5>
  <div
    id="num-stars"
  >
    <span
      id="num"
    >
      <strong>
        3.8
      </strong>
    </span>
    <div
      className="star-rating"
    >
      <div
        className="empty-stars"
      />
      <div
        className="full-stars"
        style={
          Object {
            "width": "76%",
          }
        }
      />
    </div>
  </div>
  <p>
    79
    % of reviews recommend this product
  </p>
  <div
    className="progress-bar"
  >
    <div>
      <u>
        5
         stars
      </u>
    </div>
    <div
      className="whole-bar"
    >
      <div
        className="partial-bar"
        style={
          Object {
            "width": "33%",
          }
        }
      />
    </div>
  </div>
  <div
    className="progress-bar"
  >
    <div>
      <u>
        4
         stars
      </u>
    </div>
    <div
      className="whole-bar"
    >
      <div
        className="partial-bar"
        style={
          Object {
            "width": "32%",
          }
        }
      />
    </div>
  </div>
  <div
    className="progress-bar"
  >
    <div>
      <u>
        3
         stars
      </u>
    </div>
    <div
      className="whole-bar"
    >
      <div
        className="partial-bar"
        style={
          Object {
            "width": "24%",
          }
        }
      />
    </div>
  </div>
  <div
    className="progress-bar"
  >
    <div>
      <u>
        2
         stars
      </u>
    </div>
    <div
      className="whole-bar"
    >
      <div
        className="partial-bar"
        style={
          Object {
            "width": "4%",
          }
        }
      />
    </div>
  </div>
  <div
    className="progress-bar"
  >
    <div>
      <u>
        1
         stars
      </u>
    </div>
    <div
      className="whole-bar"
    >
      <div
        className="partial-bar"
        style={
          Object {
            "width": "7%",
          }
        }
      />
    </div>
  </div>
  <div
    className="range"
  >
    <div
      className="characteristics"
    >
      <h6>
        Fit
      </h6>
      <input
        list="ticks-size"
        max="100"
        min="0"
        readOnly={true}
        type="range"
        value={53.333333333333336}
      />
      <datalist
        id="ticks-size"
      >
        <option
          value="33"
        />
        <option
          value="66"
        />
      </datalist>
    </div>
    <div
      className="characteristics"
    >
      <h6>
        Length
      </h6>
      <input
        list="ticks-size"
        max="100"
        min="0"
        readOnly={true}
        type="range"
        value={55.80645161290323}
      />
      <datalist
        id="ticks-size"
      >
        <option
          value="33"
        />
        <option
          value="66"
        />
      </datalist>
    </div>
    <div
      className="characteristics"
    >
      <h6>
        Comfort
      </h6>
      <input
        list="ticks-size"
        max="100"
        min="0"
        readOnly={true}
        type="range"
        value={57.460317460317455}
      />
      <datalist
        id="ticks-size"
      >
        <option
          value="33"
        />
        <option
          value="66"
        />
      </datalist>
    </div>
    <div
      className="characteristics"
    >
      <h6>
        Quality
      </h6>
      <input
        list="ticks-size"
        max="100"
        min="0"
        readOnly={true}
        type="range"
        value={62.53968253968254}
      />
      <datalist
        id="ticks-size"
      >
        <option
          value="33"
        />
        <option
          value="66"
        />
      </datalist>
    </div>
  </div>
</div>
`;
